src/hooks/motionWidgets/stakingWidget/useStakingInput.ts

```diff
@@ -0,0 +1,72 @@
+import { BigNumber } from 'ethers';
+import { useRef } from 'react';
+
+import {
+  getFinalStake,
+  useStakingWidgetContext,
+} from '~common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget';
+
+import { useAppContext, useColonyContext } from '~hooks';
+import { mapPayload } from '~utils/actions';
+
+import { mapStakingSliderProps } from './helpers';
+
+const useStakingInput = () => {
+  const { user } = useAppContext();
+  const { colony } = useColonyContext();
+  const mutableRef = useRef<{ limitExceeded: boolean }>();
```

Comment:
This is used to pass data between siblings in the component tree (using `useImperativeHandle` in `useStakingControls`).

`StakingControls` needs to know if the limit's been exceeded, so it can display the `ActivateButton`, but it doesn't make sense to move the `limitExceeded` state up to a shared parent in this case, because that would obligate us to hold this state in the parent every where the `StakingSlider` is used (e.g. in `ObjectionSlider` also), even if `StakingControls` doesn't also live there (as is the case in the objection modal). 

In other words, the state really does belong inside the `StakingSlider` component, and this allows it to be shared horizontally.